### DISCUSSION TITLE
How do i precisely add new menus? The documentation is insufficient on this point.

### DISCUSSION THREAD

**Discussion by bbardsley** [thuliteio/doks#645] [Q&A]
Hi there,

I wanted to create a new menu next to docs and blog, but some things still fail. I did what was said in the documentation. I've add a menu to `./config/_default/menus/menus.en.toml` and create a new folder in `./content` with some pages (all correct _index.md files). Only later I discovered that I should've also add a layout page, so I copied the doks layout folder, and gave it another name. Now I could view my new menu and it's content, but still everything does not work as expected. Somehow there is no toc even though I enabled it in every individual page. Also, the search bar does not exist. I don't understand what I'm doing wrong, because everything is just like the `docs` menu, except that it has a different name.

**Comment by h-enk** [comment]
Hi, thanks for reaching out.

Download the latest Doks — it includes PR https://github.com/h-enk/doks/pull/637 (part of the upcoming v0.4.0 release, but already available). I think this will help you 😉 

The codebase for v0.4.0 is almost ready, next is documentation.

**Reply by bbardsley** [reply]
Thank you for answering. This is a very nice command, but it illustrates precisely the problem I have. When creating a new menu in this way (with the latest version of doks), the search bar is missing in the new menu. The toc does work though.

**Reply by bbardsley** [reply]
Also, when creating a new menu in this way, the selected menu is not bold purple, as it is with docs and blog.

**Reply by h-enk** [reply]
> [..] When creating a new menu in this way (with the latest version of doks), the search bar is missing in the new menu. [..]

Currently, search only is relevant for the docs section (that's why it's in the section specific navigation bar for docs pages only — and not in the general navigation bar, or on other section pages).

Site wide search (including the option to select which sections to include) is planned for Doks v0.5.0 — see also https://github.com/h-enk/doks/issues/630

**Reply by h-enk** [reply]
> Also, when creating a new menu in this way, the selected menu is not bold purple, as it is with docs and blog.

Cannot reproduce your issue — make sure to have the menu identifiers correct.

![Snag_24222ec0](https://user-images.githubusercontent.com/3902872/152112912-45fa641e-5ab0-4330-82e5-3b539c9a7005.png)

**Reply by bbardsley** [reply]
> > Also, when creating a new menu in this way, the selected menu is not bold purple, as it is with docs and blog.
> 
> Cannot reproduce your issue — make sure to have the menu identifiers correct.
> 
> ![Snag_24222ec0](https://user-images.githubusercontent.com/3902872/152112912-45fa641e-5ab0-4330-82e5-3b539c9a7005.png)

Oh weird. I'll try fixing it then. Thanks.

**Reply by bbardsley** [reply]
I actually think the problem only occurs when there is a space in the title.

**Reply by h-enk** [reply]
Make sure to enclose `name`, `identifier`, and `parent` in "" — and use lowercase and dash in `identifier` and `parent`.

<details>
<summary>Example</summary>

```toml
[[main]]
  name = "Quick Start"
  weight = 40
  identifier = "quick-start"
  url = "/docs/prologue/quick-start/"
  parent = "get-started"
```

</details>
